base/rational.jl

```diff
@@ -564,8 +564,18 @@ end
 
 float(::Type{Rational{T}}) where {T<:Integer} = float(T)
 
-gcd(x::Rational, y::Rational) = unsafe_rational(gcd(x.num, y.num), lcm(x.den, y.den))
-lcm(x::Rational, y::Rational) = unsafe_rational(lcm(x.num, y.num), gcd(x.den, y.den))
+function gcd(x::Rational, y::Rational)
+    isinf(x) != isinf(y)
+        throw(ArgumentError("lcm is not defined between infinite and finite numbers"))
+    end
+    unsafe_rational(gcd(x.num, y.num), lcm(x.den, y.den))
+end
+function lcm(x::Rational, y::Rational)
+    isinf(x) != isinf(y)
```

Comment:
```suggestion
    if isinf(x) != isinf(y)
        throw(ArgumentError("lcm is not defined between infinite and finite numbers"))
    end
    unsafe_rational(gcd(x.num, y.num), lcm(x.den, y.den))
end
function lcm(x::Rational, y::Rational)
    if isinf(x) != isinf(y)
```
This is a typo, we're missing `if` here.